### PR TITLE
repctl to support new storage class replication SYNC and ASYNC mode

### PR DESCRIPTION
**Pull request by karthikk92**
# Description
repctl to support new storage class replication SYNC and ASYNC mode

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1443 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Create storage class with SYNC changes :
image (1)

Set mode as ASYNC and rpo as five minutes:
image (2)

Set mode as SYNC : 
image (3)

### CODE REVIEW

**Comment by nitesh3108** [comment]
pls link github issue

File: repctl/examples/powerstore_example_values.yaml [hunk 1]
```diff
@@ -15,5 +15,6 @@ parameters:
     source: "RT-A0001"
     target: "RT-A0002"
   rpo: "Five_Minutes"
+  Mode : "ASYNC"  # For SYNC rpo will be set to zero by default and not required to specify.
```
Comment (santhoshatdell) [inline repctl/examples/powerstore_example_values.yaml:18]: 'mode' as in all lower case, to be consistent with other fields.

File: repctl/pkg/cmd/templates/powerstore_source.yaml [hunk 2]
```diff
@@ -16,6 +16,7 @@ parameters:
   {{- end }}
   {{ .ReplicationPrefix }}/remoteSystem: {{ .Parameters.RemoteSystem.Source }}
   {{ .ReplicationPrefix }}/rpo: {{ .Parameters.Rpo }}
+  {{ .ReplicationPrefix }}/Mode: {{ .Parameters.Mode }}
```
Comment (santhoshatdell) [inline repctl/pkg/cmd/templates/powerstore_source.yaml:19]: Same as the previous comment for '/Mode'

File: repctl/examples/powerstore_example_values.yaml (hunk 1 above)
Comment (lukeatdell) [inline repctl/examples/powerstore_example_values.yaml:18]:
```suggestion
  Mode : "ASYNC"  # For SYNC, rpo will be set to zero by default and not required to specify.
```

**Comment by karthikk92** [comment]
> pls link github issue

Added and taken care

File: repctl/examples/powerstore_example_values.yaml (hunk 1 above)
Comment (karthikk92) [reply repctl/examples/powerstore_example_values.yaml]: taken care

Comment (karthikk92) [reply repctl/examples/powerstore_example_values.yaml]: taken care

File: repctl/pkg/cmd/templates/powerstore_source.yaml (hunk 2 above)
Comment (karthikk92) [reply repctl/pkg/cmd/templates/powerstore_source.yaml]: taken care

**Review by santhoshatdell — APPROVED** [review]

**Comment by lukeatdell** [comment]
I don't think the unit tests should be affected by these changes, but just in case, please run and report.

**Comment by lukeatdell** [comment]
> > I don't think the unit tests should be affected by these changes, but just in case, please run and report.
> 
> Its run as part of PR jobs. Let me know , so i did not run locally. image (4)

Ah yep. I must have missed the repctl test addition to the actions. 
Thank you!

**Review by lukeatdell — APPROVED** [review]

**Pull request merged**